stdlib/public/runtime/Casting.cpp

```diff
@@ -362,8 +362,24 @@ bool swift::_conformsToProtocol(const OpaqueValue *value,
 #endif
 
   
-  case MetadataKind::Existential: // FIXME
-  case MetadataKind::ExistentialMetatype: // FIXME
+  case MetadataKind::Existential: {
+#if SWIFT_OBJC_INTEROP
+    // If all protocols are @objc and at least one of them conforms to the
+    // protocol, succeed.
```

Comment:
Hm. What about non-`@objc` protocols that inherit from `@objc` protocols?

Reply:
Only `@objc` existentials can conform to protocols. A non-`@objc` protocol does not, even if it inherits from an `@objc` protocol. For example, the type checker rightly rejects this:

```swift
@objc protocol P { }
protocol Q: P { }
struct X<T:P> { }
typealias Y = X<Q>
```

Reply:
Oh, right, we aren't unboxing anything in this cast.